### DISCUSSION TITLE
Some services not working on Yi Dome Guard YRS 12.1.47* r35gb

### DISCUSSION THREAD

**Discussion by chrbar11** [roleoroleo/yi-hack-Allwinner-v2#984] [General]
Hello,
I've a Yi Dome Guard model YRS.3521, prefix SN RFUS and base version 12.1.47.02_202310111243.
I've installed r35gb_0.3.3 Yi-Hack.

Some services are working fine, like web and SSH server.
The "Disable Cloud" and "Record without Cloud" (on SD card) functions are working fine.

But streaming and snapshot are not working.
Do you have any tips to solve these problems?

I saw @Flush444's post [Dome Guard r37gb #757](https://github.com/roleoroleo/yi-hack-Allwinner-v2/discussions/757).
My Dome Guard is also the YRS.3521.
Flush444 speak about the r37gb firmware.
Is he talking about the native Yi firmware or the Yi-Hack firmware?
How can I check if my camera has the r35gb or r37gb firmware?

Thanks a lot for your help :)
Chris

![r35gb_0 3 3-status](https://github.com/user-attachments/assets/b86916d2-fd0c-4078-a3ae-e4ee26d8e6ce)

**Comment by roleoroleo** [comment]
You should open a terminal sessione t the cam and check the file /home/base/tools/extpkg.sh
The second line should be SUFFIX=...

**Reply by chrbar11** [reply]
I'm not able to open this file.
Could you please help me to do it?
```
root@test:~# cd /
root@test:/# cat /home/base/tools/extpkg.sh
root@test:/# ls
backup       etc          mnt          rdinit       sys          var
bin          home         proc         root         tmp
dev          lib          pseudo_init  sbin         usr
root@test:/# cd home
root@test:/home# ls
app      base     homever  model    qigan
root@test:/home# cd base
root@test:/home/base# ls
firmware             iso8859-1-8x16.bin   tools
iso8859-1-24x48.bin  ko                   wifi
root@test:/home/base# cd tools
root@test:/home/base/tools# ls
checkdisk    extpkg.sh    fdisk.input  fsck.fat     mkfs.fat     wpa_cli
root@test:/home/base/tools# ls -l
-rwxrwxr-x    1 root     root         11884 Oct 11  2023 checkdisk
-rw-rw-r--    1 root     root             0 Oct 11  2023 extpkg.sh
-rwxrwxr-x    1 root     root            30 Oct 11  2023 fdisk.input
-rwxrwxr-x    1 root     root         42752 Oct 11  2023 fsck.fat
-rwxrwxr-x    1 root     root         23196 Oct 11  2023 mkfs.fat
-rwxrwxr-x    1 root     root          6708 Oct 11  2023 wpa_cli
root@test:/home/base/tools#
```

**Reply by chrbar11** [reply]
> You should open a terminal sessione t the cam and check the file /home/base/tools/extpkg.sh The second line should be SUFFIX=...

Hi @roleoroleo 
Yes, it's r37gb model.
I saw you're working on it in discussion [#757](https://github.com/roleoroleo/yi-hack-Allwinner-v2/discussions/757).
I'm going to try your [beta version](https://github.com/roleoroleo/yi-hack-Allwinner-v2/discussions/757#discussioncomment-11652706).

pvi extpkg.sh file in /home/base/tools/ is empty, but I found another one at /backup/tools/extpkg.sh

Thanks :)

```
root@cameratest:~# cd /
root@cameratest:/# cd /home/base/tools
root@cameratest:/home/base/tools# ls -l
-rwxrwxr-x    1 root     root         11884 Oct 11  2023 checkdisk
-rw-rw-r--    1 root     root             0 Oct 11  2023 extpkg.sh
-rwxrwxr-x    1 root     root            30 Oct 11  2023 fdisk.input
-rwxrwxr-x    1 root     root         42752 Oct 11  2023 fsck.fat
-rwxrwxr-x    1 root     root         23196 Oct 11  2023 mkfs.fat
-rwxrwxr-x    1 root     root          6708 Oct 11  2023 wpa_cli
root@cameratest:/home/base/tools# cd /
root@cameratest:/# cd /backup/tools
root@cameratest:/backup/tools# ls -l
-rwxrwxr-x    1 root     root          4755 Oct 11  2023 default.script
-rwxr-xr-x    1 root     root          1271 Nov 16 12:33 ethdhcp.sh
-rwxrwxr-x    1 root     root          1819 Oct 11  2023 extpkg.sh
-rwxrwxr-x    1 root     root           103 Oct 11  2023 mem_update_pack.sh
-rwxrwxr-x    1 root     root        366284 Oct 11  2023 rsa_pub_dec
-rwxrwxr-x    1 root     root           639 Oct 11  2023 set_wifi.sh
-rwxrwxr-x    1 root     root           714 Oct 11  2023 update_homefs.sh
-rwxrwxr-x    1 root     root          5567 Oct 11  2023 upgrade.sh
-rwxrwxr-x    1 root     root          9052 Oct 11  2023 upgrade_firmware
-rwxrwxr-x    1 root     root           238 Oct 11  2023 wificonnect.sh
-rwxr-xr-x    1 root     root          1273 Nov 16 12:33 wifidhcp.sh
-rwxrwxr-x    1 root     root           121 Oct 11  2023 wifitest.sh
-rwxrwxr-x    1 root     root        613112 Oct 11  2023 wpa_supplicant
root@cameratest:/backup/tools# cat extpkg.sh
echo "------------------------------------home base tools extpkg.sh enter------------------------------------"
SUFFIX=r37gb
```

**Reply by roleoroleo** [reply]
Ok.
There was something wrong, your extpkg.sh was not ok.
Maybe there are other broken files?
This could be the reason of your problem.
